R/data_type.R

```diff
@@ -211,26 +213,29 @@ split_raw_data <- function(df,
   cotrt_types <- setdiff(names(df_list), "single-agent")
   
   control_sa_idx <- which(
-    rowSums(df[, conc_idx, drop = FALSE] == 0) == length(conc_idx)
+    rowSums(df[, ..conc_idx, drop = FALSE] == 0) == length(conc_idx)
   )
   control_sa <- df[control_sa_idx, ]
   untreated_tag <- gDRutils::get_env_identifiers("untreated_tag")
   
   if (length(cotrt_types) > 0) {
-    df_list[cotrt_types] <- gDRutils::loop(cotrt_types, function(x) {
-      unique_cotrt <- unique(df_list[[x]][, c(cl, drug_ids[["drug_name"]])])
+    df_list[cotrt_types] <- lapply(cotrt_types, function(x) {
```

Comment:
`gDRutils::loop` is necessary to process the data with parallel computing supported by BiocParallel